src/types/refs.rs

```diff
@@ -175,6 +194,23 @@ impl<T: ScriptClass> Drop for WeakRef<T> {
     }
 }
 
+impl<T: ScriptClass> PtrEq for WeakRef<T> {
+    #[inline]
+    fn ptr_eq(&self, other: &Self) -> bool {
+        self.0.ptr_eq(&other.0)
+    }
+}
+
+impl<T: ScriptClass> PtrEq<Ref<T>> for WeakRef<T> {
+    #[inline]
+    fn ptr_eq(&self, other: &Ref<T>) -> bool {
+        self.clone()
+            .upgrade()
+            .map(|x| x.ptr_eq(other))
+            .unwrap_or(false)
```

Comment:
upgrading here is unnecessary, it'd be better to just compare the underlying pointers directly